VAULT-4306 Ensure /raft/bootstrap/challenge call ignores erroneous namespaces set (#15519)
hashicorp/vault: changelog/15519.txt

```release-note:bug

> raft: /raft/bootstrap/challenge call now properly ignores $VAULT_NAMESPACE environment variable when set on server attempting to join a raft cluster

Reviewer: ```suggestion
raft: joining a node to a cluster now ignores any VAULT_NAMESPACE environment variable set on the server process
```
